pkg/osvscanner/osvscanner.go

```diff
@@ -842,6 +848,25 @@ func DoScan(actions ScannerActions, r reporter.Reporter) (models.VulnerabilityRe
 	return results, nil
 }
 
+// filterUnscannablePackages removes packages that don't have enough information to be scanned
+// e,g, local packages that specified by path
+func filterUnscannablePackages(packages []scannedPackage) []scannedPackage {
+	out := make([]scannedPackage, 0, len(packages))
+	for _, p := range packages {
+		switch {
+		// If none of the cases match, skip this package since it's not scannable
+		case p.Ecosystem != "" && p.Name != "" && p.Version != "":
+		case p.Commit != "":
+		case p.PURL != "":
```

Comment:
nit: maybe we should be validating that the PURL has a version as well? Or is this not possible in practice with our current extraction?

Reply:
We probably can, though we'll need to do purl parsing. I'll do this with the proper change in #644 .